pkg/systemd/init.js

```diff
@@ -823,4 +827,490 @@ define([
+        $("span", target).first().attr("value", ev.currentTarget.value);
+        switch(target.attr('id')) {
+            case "boot-or-specific-time" : set_boot_or_calendar(Number(ev.currentTarget.value));
+            break;
+            case "drop-time" : set_boot_time_unit(Number(ev.currentTarget.value));
+            break;
+            case "drop-repeat" : repeat_options(Number(ev.currentTarget.value));
+            break;
+        }
+    });
+    // Initialises create timer modal to default options.
+    function timer_init() {
+        $("#command").val("");
+        $("#description").val("");
+        $("#servicename").val("");
+        set_boot_or_calendar(1);
+        $("span", $("#boot-or-specific-time")).first().text("After system boot");
+        $("span", $("#drop-time")).first().text("Seconds");
+        $("span", $("#drop-time")).first().attr("value", "1");
+        repeat_array = [ ];
+        $(".form-control").removeClass("has-error");
+        $(".has-error").hide();
+        timer_unit = {
+            Calendar_or_Boot : "Boot",
+            boot_time_unit :"sec",
+            repeat : "Don't Repeat"
+        };
+    }
+
+    function repeat_options(val) {
+        if ($("#specific-time-error-row").is(":visible")) {
+            $("#specific-time-error-row").hide();
+            $("#hr").removeClass("has-error");
+            $("#min").removeClass("has-error");
+        }
+        switch(val) {
+            case 0 : $("#specific-time-without-repeat").show();
+                $("#repeat-time-option").hide();
+                $("#close_button").hide();
+                $("#hr").val("00");
+                $("#min").val("00");
+                timer_unit.repeat = "Don't Repeat";
+                break;
+            case 60 : timer_unit.repeat = "Repeat Hourly";// 60min
+                break;
+            case 1440 : timer_unit.repeat = "Repeat Daily";// 60*24min
+                break;
+            case 10080 : timer_unit.repeat = "Repeat Weekly";// 60*24*7min
+                break;
+            case 44640 : timer_unit.repeat = "Repeat Monthly";// 60*24*31min
+                break;
+            case 525600 : timer_unit.repeat = "Repeat Yearly";// 60*24*365min
+                break;
+        }
+        if (val) {
+            $("#specific-time-without-repeat").hide();
+            $("#repeat-time-option").show();
+            repeat_array = [ ];
+            repeat_element();
+        }
+    }
+
+    function repeat_element() {
+        var close = "enabled";
+        if (repeat_array.length === 0)
+            close = "disabled";
+
+        var repeat_contents = {
+            index:repeat_array.length,
+            close:close,
+            hours:"00",
+            minutes:"00",
+            days_value:"1",
+            days_text:"Monday",
+            date_to_parse:new Date(),
+            date:moment().format("YYYY-MM-DD")
+        };
+
+        if (timer_unit.repeat == "Repeat Hourly") {
```

Comment:
Many of the blocks in this `if-else` construct look similar. Can they be combined? You could use `||` in the conditions.